src/box/tuple.h

```diff
@@ -356,9 +357,55 @@ enum {
 	TUPLE_REF_MAX = UINT8_MAX,
 };
 
+/**
+ * Check that data_offset is valid and can be stored in tuple.
+ * @param data_offset - see member description in struct tuple.
+ *
+ * @return 0 on success, -1 otherwise (diag is set).
+ */
+static inline int
+tuple_check_data_offset(uint32_t data_offset)
+{
+	if (data_offset > INT16_MAX) {
+		/** tuple->data_offset is 15 bits */
```

Comment:
Not after the previous commit. Please fix. Preferably, in the previous commit.

Reply:
It was, is and will be limited with 15 bits. In the next commit the corresponding high bit will be used for `is_compact` flag. There's no need to allow 16 bits and immediately prohibit back again.